crates/store/src/server/api.rs

```diff
@@ -220,6 +220,45 @@ impl api_server::Api for StoreApi {
         }))
     }
 
+    /// Returns info which can be used by the client to sync note state.
+    #[instrument(
+        target = "miden-store",
+        name = "store:sync_notes",
+        skip_all,
+        ret(level = "debug"),
+        err
+    )]
+    async fn sync_notes(
+        &self,
+        request: tonic::Request<SyncNoteRequest>,
+    ) -> Result<Response<SyncNoteResponse>, Status> {
+        let request = request.into_inner();
+
+        let (state, delta) = self
+            .state
+            .sync_state(request.block_num, &[], &request.note_tags, &[])
+            .await
+            .map_err(internal_error)?;
```

Comment:
While this works, I think we should have a dedicated method which does not make extra requests to the database or maybe modify the existing method to avoid extra requests if `account_ids` and `nullifier_prefixes` are not provided. Could be done in a follow-up PR though.

Reply:
I've implemented this with all of the lower down pieces - I was surprised at how many parts there were to add and invoke a single db query.

Reply:
Let's create an issue to refactor this.